### ISSUE TITLE
Test NVTabular, Petastorm, and Huggingface Datasets for parquet data loading

### ISSUE THREAD

**Issue by zhujiem** [reczoo/FuxiCTR#88]
Huggingface Datasets:
```
 dataset = load_dataset("parquet", data_files={split: data_blocks}, split=split)
 super().__init__(dataset=dataset, num_workers=8, batch_size=self.batch_size)
```

**Comment by xpai** [comment]
After comparing the three options, we chose **HuggingFace `datasets`** as the parquet loader backend — NVTabular and Petastorm have been unmaintained for a long time, while `datasets` is actively maintained, has a stable `load_dataset("parquet", ...)` API, and integrates with `torch.utils.data.DataLoader`.

Implemented in `15ddfc6`:
- New `fuxictr/pytorch/dataloaders/hf_dataloader.py`: `HFDataLoader` wraps `datasets.load_dataset("parquet", ...)` with a torch `DataLoader` for batching/shuffle/multi-worker; supports streaming (`IterableDataset` + buffered shuffle) and reads sample count from the parquet footer (no cache).
- `RankDataLoader`: `data_format="parquet"` routes to `HFDataLoader`; removed obsolete `parquet_block_dataloader.py` / `parquet_dataloader.py`.
- Migrated `tiny_seq` sample data npz → parquet; updated 7 `dataset_config.yaml` to `data_format: parquet`.

Follow-up `d7c3834` fixed the downstream parquet dtype: encoded int IDs were being written as `str` by `datasets.map` on a from_polars dataset, so `build_dataset.transform` now does `Dataset.cast(Features)` to restore int64/float64 and `with_format("torch")` yields tensors.

Closing as resolved.

**Comment by zhujiem** [comment]
After production testing, we found HuggingFace `datasets` was significantly slower than the original numpy-based dataloader — roughly **10x slower** in data loading throughput. This performance regression was unacceptable for training on large-scale datasets.

To fix this, we replaced the HuggingFace `datasets` backend with **Ray Data** (`ray.data`), which achieves throughput comparable to the original numpy version while keeping the same `RankDataLoader` API contract. The key changes include:

- Replaced `HFDataLoader` with `ParquetDataLoader` using `ray.data.read_parquet` + `iter_torch_batches` with `local_shuffle_buffer_size` for per-epoch row-level shuffle.
- Migrated `build_dataset.transform()` from `datasets.map` to `ray.data.map_batches` for feature encoding.
- Removed the `datasets` (HuggingFace) dependency entirely.

This was implemented in `e701804` and further refined in subsequent commits. Released in **v2.3.10**.
